pkg/ovirt/ovirt.es6

```diff
@@ -0,0 +1,286 @@
+    downloadIcons,
+} from './actions.es6';
+
+import { logDebug, logError } from '../machines/helpers.es6';
+import { ovirtApiGet } from './ovirtApiAccess.es6';
+import CONFIG from './config.es6';
+import { isOvirtApiCheckPassed } from './provider.es6';
+
+let lastOvirtPoll = -1; // timestamp
+/**
+ * Initiate polling of oVirt data.
+ *
+ * @param dispatch
+ */
+export function startOvirtPolling({ dispatch }) {
+    if (!isOvirtApiCheckPassed()) {
+        logDebug(`Skipping oVirt poling due to failed/unfinished oVirt API check`);
+        return;
+    }
+
+    window.setInterval(() => dispatch(pollOvirtAction()), 1000);
+}
+
+/**
+ * Ensure, the function `call()` is not executed more then once per timeperiod.
+ */
+function callOncePerTimeperiod({ call, delay, lastCall }) {
+    const now = Date.now();
+
+    if (lastCall + delay <= now) {
+        return call();
+    }
+
+    return null;
+}
+
+export function pollOvirt() {
+    return function (dispatch) {
+        callOncePerTimeperiod({
+            lastCall: lastOvirtPoll,
+            delay: CONFIG.ovirt_polling_interval, // do not poll more than once per this period
+            call: () => {
+                if (cockpit.hidden) { // plugin is not visible in Cockpit
+                    logDebug('oVirt polling skipped since the plugin is not visible');
+                    return;
+                }
+
+                logDebug('Executing oVirt polling');
+                lastOvirtPoll = Infinity; // avoid parallel execution
+                const promises = [];
+                promises.push(doRefreshHosts(dispatch));
+                promises.push(doRefreshVms(dispatch));
+                promises.push(doRefreshTemplates(dispatch));
+                promises.push(doRefreshClusters(dispatch));
+
+                return Promise.all(promises).then(() => { // update the timestamp
+                    logDebug('oVirt polling finished');
+                    lastOvirtPoll = Date.now(); // single polling finished, re-enable it
+                });
+            }
+        });
+    };
+}
+
+/**
+ * Shortens the period for next oVirt polling, so it will be executed at next earliest opportunity.
+ *
+ * Useful to shorten polling delay after user action.
+ */
+export function forceNextOvirtPoll() {
+    lastOvirtPoll = -1;
+}
+
+function doRefreshHosts(dispatch) {
+    logDebug(`doRefreshHosts() called`);
+    return ovirtApiGet('hosts').done(result => {
+        if (result && result.host && (result.host instanceof Array)) {
+            const allHostIds = [];
+            result.host.forEach( host => {
+                allHostIds.push(host.id);
+                dispatch(updateHost({
+                    id: host.id,
+                    name: host.name,
+                    address: host.address,
+                    clusterId: host.cluster ? host.cluster.id : undefined,
+                    status: host.status,
+                    memory: host.memory,
+                    cpu: host.cpu ? {
```

Comment:
Why would we refresh host cpu name, speed, etc.? Is it ever going to change after the 1st refresh?